index.js

```diff
@@ -384,7 +384,78 @@ Color.prototype = {
 				w1 * color1.green() + w2 * color2.green(),
 				w1 * color1.blue() + w2 * color2.blue(),
 				color1.alpha() * p + color2.alpha() * (1 - p));
+	},
+
+	gradientRGB: function (toColor, steps) {
+		if (!toColor || !toColor.rgb) {
+			throw new Error('Argument "toColor" was not a Color instance, but rather an instance of ' + typeof toColor);
+		}
+		if (steps < 2) {
+			throw new Error('Argument "steps" must to be bigger or equal to 2.');
+		}
+		toColor = toColor.rgb();
+		var fromColor = this.rgb();
+		var [fR, fG, fB] = fromColor.color;
+		var [tR, tG, tB] = toColor.color;
+		var fA = fromColor.alpha();
+		var tA = toColor.alpha();
+		var incR = (tR - fR) / (steps - 1);
+		var incG = (tG - fG) / (steps - 1);
+		var incB = (tB - fB) / (steps - 1);
+		var incA = (tA - fA) / (steps - 1);
+		var gradient = [fromColor];
+		for (var i = 1; i < steps; i++) {
+			gradient.push(Color({
+				r: Math.round(fR + incR * i),
+				g: Math.round(fG + incG * i),
+				b: Math.round(fB + incB * i),
+				alpha: fA + incA * i
+			}));
+		}
```

Comment:
"save a bunch of branching cycles", nice!

I believe the color component must to be an one byte int. However if it suports a flot, great! It means less error on conversions. I'll remove the `round()`.

Reply:
There's nothing that says it has to be a single byte - in fact, that's not even enforceable in JavaScript. :)

Reply:
The imageData from canvas is Uint8ClampedArray, as in most color use cases, that is why i think. But you are right makeing it free from this limitation.

Reply:
> as in most color use cases

Most prominent? Perhaps, as 8bpp images are the most frequently used in web. But `color` doesn't assume the application. MacOS uses floating point channels, TIFF uses whatever you want (including crazy stuff like packed 13 bits per channel and 1000 channels in an image), etc.

Color doesn't, and shouldn't, have an opinion about that. If your application needs clamped and truncated, then you'll have to do it. `Math.round` in and of itself is an opinion, too - it could be `Math.ceil` or `Math.floor` depending on the application.

Hope that helps :)